app/controllers/editions_controller.rb

```diff
@@ -34,8 +39,32 @@ def linking
     render action: "show"
   end
 
-  def unpublish
-    render action: "show"
+  def confirm_unpublish
+    if redirect_url.blank? || validate_redirect(redirect_url)
+      render "secondary_nav_tabs/confirm_unpublish"
+    else
+      error_message = "Redirect path is invalid. #{description(resource)} can not be unpublished."
+      @resource.errors.add(:redirect_url, error_message)
+      render "show"
+    end
+  end
+
+  def process_unpublish
+    artefact = @resource.artefact
+
+    success = params["redirect_url"].strip.empty? ? UnpublishService.call(artefact, current_user) : UnpublishService.call(artefact, current_user, redirect_url)
```

Comment:
For readability  - maybe worth cleaning it up a little, setting `success` to a method defined outside of `process_unpublish` method.